### ISSUE TITLE
Session stick anotation in ingress no work

### ISSUE THREAD

**Issue by Itiho** [rancher/rancher#13702]
**Rancher versions:**
rancher/rancher:v2.0.1

**Docker version:
Containers: 32
 Running: 28
 Paused: 0
 Stopped: 4
Images: 26
Server Version: 18.03.1-ce
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 246
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 773c489c9c1b21a6d78b5c538cd395416ec50f88
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.93-boot2docker
Operating System: Boot2Docker 18.03.1-ce (TCL 8.2.1); HEAD : cb77972 - Thu Apr 26 16:40:36 UTC 2018
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 3.858GiB
Name: k8s-ld-master-1
ID: BGLM:PGCY:H2Y4:CZWE:T7YB:P2NM:YJAP:BMSG:LYD4:KILX:2TWX:BMHK
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
 provider=vmwarevsphere
Experimental: false
Insecure Registries:
 127.0.0.0/8
Registry Mirrors:
Live Restore Enabled: false

**Operating system and kernel:boot2docker 18.03.1-ce

**Type/provider of hosts: vSphere

**Setup details: single node rancher

**Environment Template: Kubernetes
**Steps to Reproduce:**

-Create a cluster
-Create a deployment with image tutum/hello-world (to see pods information)
-Create a service
-Create a ingress with annotations
    nginx.ingress.kubernetes.io/affinity: "cookie"
    nginx.ingress.kubernetes.io/session-cookie-name: "route"
    nginx.ingress.kubernetes.io/session-cookie-hash: "sha1"
-Accessing the URL there is alternation between the PODs

Change the "nginx-ingress-controller" deployment image to **quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.15.0** and manual delete the pods (no automatic update) the stick session works fine

**Comment by Itiho** [comment]
If i use / in path works, but in official image this is'nt necessary

**Comment by loganhz** [comment]
It still happens in v2.0.6

**Comment by orangedeng** [comment]
@loganhz @Itiho I think we fix it with next version of controller. `rancher/nginx-ingress-controller:0.16.2-rancher1` would be fine. It is already applied to 1.9.7, 1.10.5 and 1.11.1 cluster in next release.

**Comment by alena1108** [comment]
@orangedeng could you put the link to corresponding fix/pr?

**Comment by sangeethah** [comment]
Tested with latest build from master:
Create a workload.
Create an ingress pointing to the above workload with the following annotaions:

```
nginx.ingress.kubernetes.io/affinity: cookie
nginx.ingress.kubernetes.io/session-cookie-name: route
nginx.ingress.kubernetes.io/session-cookie-hash: sha1
```
Note - There should be no double quotes specified for the values.

When accessing ingress url from browser , it directs traffic to only instance as expected.


Ingress gets programmed with sticky options as expected.

/etc/nginx/nginx.conf :

```
        upstream sticky-default-ingress-bf95e7f08fa77dde6a62552e758378f3-80 {
                sticky hash=sha1 name=route path=/ httponly;

                keepalive 32;

                server 10.42.3.33:80 max_fails=0 fail_timeout=0;
                server 10.42.2.28:80 max_fails=0 fail_timeout=0;
                server 10.42.2.27:80 max_fails=0 fail_timeout=0;

        }

        upstream default-ingress-bf95e7f08fa77dde6a62552e758378f3-80 {
                least_conn;

                keepalive 32;

                server 10.42.3.33:80 max_fails=0 fail_timeout=0;
                server 10.42.2.28:80 max_fails=0 fail_timeout=0;
                server 10.42.2.27:80 max_fails=0 fail_timeout=0;

        }
```

**Comment by sangeethah** [comment]
This fix is available from 2.0.7 release that uses rancher/nginx-ingress-controller:0.16.2-rancher1.
